includes/wc-payment-api/class-wc-payments-api-client.php

```diff
@@ -607,11 +607,18 @@ public function list_transactions( $page = 0, $page_size = 25, $sort = 'date', $
 
 		$transactions = $this->request( $query, self::TRANSACTIONS_API, self::GET );
 
+		$charge_ids             = array_column( $transactions['data'], 'charge_id' );
+		$orders_with_charge_ids = $this->wcpay_db->orders_with_charge_id_from_charge_ids( $charge_ids );
+
 		// Add order information to each transaction available.
 		// TODO: Throw exception when `$transactions` or `$transaction` don't have the fields expected?
 		if ( isset( $transactions['data'] ) ) {
 			foreach ( $transactions['data'] as &$transaction ) {
```

Comment:
Let's add `unset($transaction);` after the loop with reference to secure future changes from accidentally modifying the reference content.